src/iceberg/type.h

```diff
@@ -78,20 +79,23 @@ class ICEBERG_EXPORT NestedType : public Type {
   /// \brief Get a field by field ID.
   ///
   /// \note This is O(1) complexity.
-  [[nodiscard]] virtual std::optional<std::reference_wrapper<const SchemaField>>
+  [[nodiscard]] virtual Result<std::optional<std::reference_wrapper<const SchemaField>>>
```

Comment:
nit: `using SchemaFieldConstRef = std::reference_wrapper<const SchemaField>;` to make it shorter and replace all return types below?